openwhisk/runHandler.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openwhisk
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// Params are the parameteres sent to the action
+type Params struct {
+	Value json.RawMessage `json:"value"`
+}
+
+// ErrResponse is the response when there are errors
+type ErrResponse struct {
+	Error string `json:"error"`
+}
+
+func sendError(w http.ResponseWriter, code int, cause string) {
+	errResponse := ErrResponse{Error: cause}
+	b, err := json.Marshal(errResponse)
+	if err != nil {
```

Comment:
What happens if this error actually happens? What will the value of `b` be when it is written below?

Reply:
Actually this code comes from James Thomas initial implementation I used as a basis. However since we are marshalling a well known data structure, it cannot happen it fails. But if it would, a <nil> message would be generated.